modules/core/resources/META-INF/smithy/datetime.smithy

```diff
@@ -0,0 +1,186 @@
+
+@offsetDateTimeFormat
+@timestampFormat("date-time")
+timestamp OffsetDateTime
+
+
+/// This trait indicates that a String contains a time value with an offset in
+/// the ISO-8601 format HH:MM:SS[.s{1,9}](+|-)HH:MM
+/// Fractional seconds support a precision up to the nanosecond.
+@trait(selector: ":test(string, member > string)")
+structure offsetTimeFormat { }
+
+@offsetTimeFormat
+string OffsetTime
+
+
+/// This trait indicates that a String contains a timezone identifier such as
+/// America/New_York or an offset from UTC such as +01:00
+@trait(selector: ":test(string, member > string)")
+structure zoneIdFormat { }
+
+@zoneIdFormat
+string ZoneId
+
+
+/// This triat indicates that a String contains a timezone offset from UTC
+/// such as +01:00
+@trait(selector: ":test(string, member > string)")
+structure zoneOffsetFormat { }
+
+@zoneOffsetFormat
+string ZoneOffset
+
+
+/// This trait indicates that a String contains a time value with an offset and
+/// a timezone identifier in the ISO-8601 format
+/// YYYY-mm-DDTHH:MM:SS[.s{1,9}](+|-)HH:MM\[ZONEID]
+/// Fractional seconds support a precision up to the nanosecond.
+@trait(selector: ":test(string, member > string)")
+structure zonedDateTimeFormat { }
+
+@zonedDateTimeFormat
+string ZonedDateTime
+
+enum DayOfWeek {
+  MONDAY
+  TUESDAY
+  WEDNESDAY
+  THURSDAY
+  FRIDAY
+  SATURDAY
+  SUNDAY
+}
+
+
+enum Month {
+  JANUARY
+  FEBRUARY
+  MARCH
+  APRIL
+  MAY
+  JUNE
+  JULY
+  AUGUST
+  SEPTEMBER
+  OCTOBER
+  NOVEMBER
+  DECEMBER
+}
+
+
+/// A year in the ISO-8601 format.
+@trait(selector: ":test(integer, member > integer)")
+structure yearFormat { }
+
+@yearFormat
+integer Year
+
+
+/// This trait indicates that a String contains a value describng a year and
+/// month in the ISO-8601 format.
+/// Example: 2025-05 reads as May of year 2025
+@trait(selector: ":test(string, member > string)")
+structure yearMonthFormat { }
+
+@yearMonthFormat
+string YearMonth
+
+
+/// This trait indicates that a String contains a describint a month and day
+/// in the ISO-8601 format mm-DD
+/// Example: "05-14" reads as May 14th
+@trait(selector: ":test(string, member > string)")
+structure monthDayFormat { }
+
+@monthDayFormat
+string MonthDay
+
+
+/// This trait indicates that a String contains a time duration value based on
+/// the ISO-8601 duration format PnDTnHnMn.nS with days considered to be
+/// exactly 24 hours.
+/// For full format details see https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/time/Duration.html#parse(java.lang.CharSequence)
+///
+/// Examples:
+/// * "PT15.784S" reads as 15 seconds and 784 milliseconds
+/// * "PT25M" reads as 25 minutes
+/// * "PT6H" reads as 6 hours
+/// * "PT-6H3M" reads as -6 hours and +3 minutes
+/// * "-PT6H3M" reads as -6 hours and -3 minutes
+/// * "-PT-6H+3M" reads as +6 hours and -3 minutes
+@trait(selector: ":test(string, member > string)")
+structure durationFormat { }
+
+@durationFormat
+string Duration
```

Comment:
actually, I'm not too sure about supporting this one, as the spec does nothing to disambiguate exactly what it means for a duration : https://en.wikipedia.org/wiki/ISO_8601#Durations 

Implementation-wise (in smithy4s), I want durations to be represented like timestamps (as as seconds/nanoseconds). We cannot realistically the Iso8601 duration format into such a representation, because of the ambiguity.

Reply:
I'm removing the `Duration` and `Period` definitons and we can decide at a later time how we want to express these concepts in alloy.